src/Murder.Editor/Data/EditorDataManager.cs

```diff
@@ -257,20 +257,33 @@ private void LoadEditorSettings()
                 _gameProfile = FileHelper.DeserializeAsset<GameProfile>(gameProfilePath)!;
             }
 
-            // Create a game profile, if none was provided from the base game or if the game
-            // provides a different one.
-            // TODO: Is there a better way to verify if the profile match?
-            GameProfile profile = CreateGameProfile();
-            if (_gameProfile is null || _gameProfile.GetType() != profile.GetType())
+            // // Create a game profile, if none was provided from the base game or if the game
+            // // provides an incompatible one.
+            if (!ValidateGameProfileType())
             {
                 GameLogger.Warning($"Didn't find {GameProfileFileName} file. Creating one.");
-
+                GameProfile profile = CreateGameProfile();
                 GameProfile = profile;
                 GameProfile.MakeGuid();
                 SaveAsset(GameProfile);
             }
         }
 
+        private bool ValidateGameProfileType()
+        {
+            if (_gameProfile == null)
```

Comment:
(super nit) I usually prefer using `is` for null checks, since a type can override the `==` operator